osquery/core/watcher.cpp

```diff
@@ -322,9 +322,9 @@ void WatcherRunner::createWorker() {
       LOG(WARNING) << "osqueryd worker respawning too quickly: "
                    << Watcher::workerRestartCount() << " times";
       Watcher::workerRestarted();
-      interruptableSleep(getWorkerLimit(RESPAWN_DELAY) * 1000);
+      pauseMilli(getWorkerLimit(RESPAWN_DELAY) * 1000);
       // Exponential back off for quickly-respawning clients.
-      interruptableSleep(pow(2, Watcher::workerRestartCount()) * 1000);
+      pauseMilli(pow(2, Watcher::workerRestartCount()) * 1000);
```

Comment:
Combine these two pauses. Otherwise an interrupted initial pause will except/catch into the second if interrupted.